react-components/src/tree-view/view/components/tree-view-caret.test.tsx

```diff
@@ -0,0 +1,59 @@
+import { describe, expect, test } from 'vitest';
+import userEvent from '@testing-library/user-event';
+import { render } from '@testing-library/react';
+import { getIconName, getIconsInContainer } from '#test-utils/cogs/htmlTestUtils';
+import { TreeNode } from '../../model/tree-node';
+import { TreeViewCaret, type TreeViewCaretProps } from './tree-view-caret';
+
+describe(TreeViewCaret.name, () => {
+  test('should not render the caret when it is leaf', () => {
+    const node = new TreeNode();
+
+    const container = renderMe({ node });
```

Comment:
Optional / suggestion:


Personally I would just do this:

```suggestion
    const { container } = render(<TreeViewCaret node={node} />);
```

There is almost no code saved from using a common `renderMe` function. And personally I find it easier to read and understand when the actual implementation is just there. Extending the test becomes harder in the current solution. If I f.ex. want to access the `rerender` function from the `render` utility, I now need to change all tests using the `renderMe` function.

Reply:
I just followed the same pattern everywhere for those element. I think I will leave it as it is. But I see you point